internal/output/progress.go

```diff
@@ -96,3 +100,7 @@ func (d *Spinner) Stop(msg string) {
 
 	d.out.Fprint(d.out.Config().ErrWriter, "\n")
 }
+
+func isInteractive(out Outputer) bool {
+	return out.Config().Interactive && terminal.IsTerminal(int(os.Stdin.Fd())) && (runtime.GOOS != "windows" || os.Getenv("SHELL") != "")
```

Comment:
We can't rely on `runtime.GOOS` as Windows also has bash shells. Windows and cmd.exe are not the same.

Reply:
FYI: that's what the `os.Getenv("SHELL") != ""` is for. Bash on Windows will have a non-empty "SHELL" env.

Reply:
> FYI: that's what the `os.Getenv("SHELL") != ""` is for. Bash on Windows will have a non-empty "SHELL" env.

Oh I had not caught onto that, that's clever! Still, I'd rather we centralized this behavior in subshell. Such that wherever we detect the subshell it's always the same mechanic.